cooperative_lanechange/src/cooperative_lanechange_node.cpp

```diff
@@ -0,0 +1,664 @@
+        
+    lanelet::LaneletMapConstPtr const_map(wm_->getMap());
+    lanelet::ConstLanelet veh2_lanelet = const_map->laneletLayer.get(veh2_lanelet_id);
+    RCLCPP_DEBUG_STREAM(get_logger(), "veh2_lanelet id " << veh2_lanelet.id());
+
+    auto current_lanelets = lanelet::geometry::findNearest(const_map->laneletLayer, ego_pos, 10);       
+    if(current_lanelets.size() == 0)
+    {
+      RCLCPP_WARN_STREAM(get_logger(), "Cannot find any lanelet in map!");
+      return true;
+    }
+    lanelet::ConstLanelet current_lanelet = current_lanelets[0].second;
+    RCLCPP_DEBUG_STREAM(get_logger(), "current llt id " << current_lanelet.id());
+        
+    //Create temporary route between the two vehicles
+    lanelet::ConstLanelet start_lanelet = veh2_lanelet;
+    lanelet::ConstLanelet end_lanelet = current_lanelet;
+        
+    auto map_graph = wm_->getMapRoutingGraph();
+    RCLCPP_DEBUG_STREAM(get_logger(), "Graph created");
+
+    auto temp_route = map_graph->getRoute(start_lanelet, end_lanelet);
+    RCLCPP_DEBUG_STREAM(get_logger(), "Route created");
+
+    //Throw exception if there is no shortest path from veh2 to subject vehicle
+    lanelet::routing::LaneletPath shortest_path2;
+    if(temp_route)
+    {
+      shortest_path2 = temp_route.get().shortestPath();
+    }
+    else{
+      RCLCPP_ERROR_STREAM(get_logger(), "No path exists from roadway object to subject");
+      throw std::invalid_argument("No path exists from roadway object to subject");
+    }
+ 
+    RCLCPP_DEBUG_STREAM(get_logger(), "Shorted path created size: " << shortest_path2.size());
+    for (auto llt : shortest_path2)
+    {
+      RCLCPP_DEBUG_STREAM(get_logger(), "llt id  route: " << llt.id());
+    }  
+
+    //To find downtrack- creating temporary route from veh2 to veh1(ego vehicle)
+    double veh1_current_downtrack = wm_->routeTrackPos(ego_pos).downtrack;      
+    RCLCPP_DEBUG_STREAM(get_logger(), "ego_current_downtrack:" << veh1_current_downtrack);
+        
+    current_gap = veh1_current_downtrack - veh2_downtrack;
+    RCLCPP_DEBUG_STREAM(get_logger(), "Finding current gap");
+    RCLCPP_DEBUG_STREAM(get_logger(), "Veh1 current downtrack: " << veh1_current_downtrack << " veh2 downtrack: " << veh2_downtrack);
+     
+    return current_gap;
+  }
+
+  void CooperativeLaneChangePlugin::pose_cb(const geometry_msgs::msg::PoseStamped::UniquePtr msg)
+  {
+    pose_msg_ = *msg;
+  }
+  
+  void CooperativeLaneChangePlugin::twist_cb(const geometry_msgs::msg::TwistStamped::UniquePtr msg)
+  {
+    current_speed_ = msg->twist.linear.x;
+  }
+    
+  void CooperativeLaneChangePlugin::bsm_cb(const carma_v2x_msgs::msg::BSM::UniquePtr msg)
+  {
+    bsm_core_ = msg->core_data;
+  }
+  
+  void CooperativeLaneChangePlugin::plan_trajectory_callback(
+    std::shared_ptr<rmw_request_id_t>, 
+    carma_planning_msgs::srv::PlanTrajectory::Request::SharedPtr req, 
+    carma_planning_msgs::srv::PlanTrajectory::Response::SharedPtr resp)
+  {
+    //@SONAR_STOP@
```

Comment:
Can you take this SONAR_STOP and SONAR_START out. This is way too much actual code to disable analysis on.

Reply:
Fixed.